Run tests and releases via Github Actions

theskumar opened the pull request:
- [ ] coveralls integration

bbc2 commented on .github/workflows/release.yml:
```diff
@@ -0,0 +1,25 @@
+        TWINE_USERNAME: ${{ secrets.PYPI_USERNAME }}
+        TWINE_PASSWORD: ${{ secrets.PYPI_PASSWORD }}
+      run: |
+        make release
```
This looks good but isn't it going to fail if the package version was already uploaded by Travis?

theskumar replied:
As this action is only supposed to run when a new tag( release) is created. I'm hoping this won't conflict with existing versions on PyPi.

bbc2 replied:
Travis also uploads packages when a tag is created, so if it uploads before the GitHub Action, then `twine` may return with an error.

In any case, I'm happy to merge this and try it if you want.  The GitHub Action will most likely be executed first since it's faster than Travis CI.  We could also disable the Travis CI deploy job.

bbc2 reviewed (commented):
Looks rather good to me. I really like GitHub actions.

theskumar commented:
@bbc2 I've adjusted few things to work it well with coverall. 

I've removed travis.

Note the coverage is now report against each of the env and then an average of all the reported. On travis coverage was sent for just one env.

There is an issue with new setup in which the files on coverall doesn't map correctly with the source code GitHub. As the files reported now live within `.tox/py*/lib/` folder. This is probably due to `AndreMiras/coveralls-python-action@develop` GitHub action I'm using, but it shouldn't be much of an issue right now.

bbc2 commented on tox.ini:
```diff
@@ -1,5 +1,16 @@
-envlist = lint,py{27,34,35,36,37,38,39,34-no-typing},pypy,pypy3,manifest,coverage-report
+envlist = lint,py{27,35,36,37,38,39},pypy,pypy3,manifest,coverage-report
+
+[gh-actions]
+python =
+    2.7: py27, coverage-report
+    3.5: py35, coverage-report
+    3.6: py36, coverage-report
+    3.7: py37, coverage-report
+    3.8: py38, coverage-report
+    3.9: py39, mypy, lint, manifest, coverage-report
+    pypy2: pypy, coverage-report
+    pypy3: pypy3, coverage-report
```
Do we really need the `coverage-report` environment anymore?  My impression is that it creates duplicate Coveralls reports in PRs.  I think this is not necessary since the GitHub action you added is supposed to communicate with coveralls and the only thing we need to do is to run the tests with coverage enabled (which is already the case for the `py27`, `py35`, ..., environments.

bbc2 commented:
I've removed the integration with coveralls because I think it's too confusing to be useful, especially to new contributors. For instance, there are too many reports in the GitHub results and too many false positives:

![coveralls_results](https://user-images.githubusercontent.com/1645335/108605742-6fb05000-73b6-11eb-8937-3fe6b5138136.png)

Also, like you pointed out, there seem to be an issue in how the reports for different versions are combined and the path of source files:

![coveralls_report](https://user-images.githubusercontent.com/1645335/108605764-95d5f000-73b6-11eb-908a-78c7fb9ee32d.png)

So, I'll merge the actions because it's nice to have a CI and I'd like to move forward with the package, but I'll just leave out Coveralls for now.  I suggest we reconsider it later. I hope you don't mind.
